lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -38,15 +38,24 @@ def as_json
 
         private
 
-        def data_for(serializer)
-          if serializer.respond_to?(:each)
+        def data_for(association)
+          if belongs_to_with_foreign_key?(association)
+            id = @object.send(association.reflection.foreign_key)
+            type = association.reflection.type.to_s
```

Comment:
that would be a good change too.

what I was kind of thinking is abstracting out more methods for the conditions to read more like what the the functionality is:

like, on line 46, maybe: `association.is_collection?`

and then the building of the ResourceIdentifier(s, opt).as_json could be its own method, since serializable_resource_options are outside the scope of _this_ method. maybe:

```ruby
def jsonized_resource_for(serializer)
  ResourceIdentifier.new(serializer, serializable_resource_options).as_json
end
```

idk

that also allows us to test the logic of `data_for` without worrying about if the implementation of `ResourceIdentifier` is correct.